code/modules/jobs/job_exp.dm

```diff
@@ -0,0 +1,203 @@
+	if(!exp_read.Execute())
+		var/err = exp_read.ErrorMsg()
+		log_game("SQL ERROR during exp_update_client read. Error : \[[err]\]\n")
+		message_admins("SQL ERROR during exp_update_client read. Error : \[[err]\]\n")
+		return
+	var/list/read_records = list()
+	while(exp_read.NextRow())
+		read_records[EXP_TYPE_LIVING] = exp_read.item[1]
+		read_records[EXP_TYPE_CREW] = exp_read.item[2]
+		read_records[EXP_TYPE_COMMAND] = exp_read.item[3]
+		read_records[EXP_TYPE_ENGINEERING] = exp_read.item[4]
+		read_records[EXP_TYPE_MEDICAL] = exp_read.item[5]
+		read_records[EXP_TYPE_SCIENCE] = exp_read.item[6]
+		read_records[EXP_TYPE_SUPPLY] = exp_read.item[7]
+		read_records[EXP_TYPE_SECURITY] = exp_read.item[8]
+		read_records[EXP_TYPE_SILICON] = exp_read.item[9]
+		read_records[EXP_TYPE_SERVICE] = exp_read.item[10]
+		read_records[EXP_TYPE_SPECIAL] = exp_read.item[11]
+		read_records[EXP_TYPE_GHOST] = exp_read.item[12]
+		read_records[EXP_TYPE_EXEMPT] = exp_read.item[13]
+	var/list/play_records = list()
+	for(var/rtype in GLOB.exp_jobsmap)
+		if(text2num(read_records[rtype]))
+			play_records[rtype] = text2num(read_records[rtype])
+		else
+			play_records[rtype] = 0
+	if(mob.stat == CONSCIOUS && mob.mind.assigned_role)
+		play_records[EXP_TYPE_LIVING] += minutes
+		if(announce_changes)
+			to_chat(mob,"<span class='notice'>You got: [minutes] Living EXP!")
+		for(var/category in GLOB.exp_jobsmap)
+			if(GLOB.exp_jobsmap[category]["titles"])
+				if(mob.mind.assigned_role in GLOB.exp_jobsmap[category]["titles"])
+					play_records[category] += minutes
+					if(announce_changes)
+						to_chat(mob,"<span class='notice'>You got: [minutes] [category] EXP!")
+		if(mob.mind.special_role)
+			play_records[EXP_TYPE_SPECIAL] += minutes
+			if(announce_changes)
+				to_chat(mob,"<span class='notice'>You got: [minutes] Special EXP!")
+	else if(isobserver(mob))
+		play_records[EXP_TYPE_GHOST] += minutes
+		if(announce_changes)
+			to_chat(mob,"<span class='notice'>You got: [minutes] Ghost EXP!")
+	else if(minutes != 0)	//Let "refresh" checks go through
+		return
+	prefs.exp = play_records
+	var/datum/DBQuery/update_query = SSdbcore.NewQuery("INSERT INTO [format_table_name("role_time")] (`ckey`) VALUES ('[ckey]') ON DUPLICATE KEY UPDATE  living = '[play_records[EXP_TYPE_LIVING]]', crew ='[play_records[EXP_TYPE_CREW]]', command = '[play_records[EXP_TYPE_COMMAND]]', engineering = '[play_records[EXP_TYPE_ENGINEERING]]', medical = '[play_records[EXP_TYPE_MEDICAL]]', science = '[play_records[EXP_TYPE_SCIENCE]]', supply = '[play_records[EXP_TYPE_SUPPLY]]', security = '[play_records[EXP_TYPE_SECURITY]]', silicon = '[play_records[EXP_TYPE_SILICON]]', service ='[play_records[EXP_TYPE_SERVICE]]', special = '[play_records[EXP_TYPE_SPECIAL]]', ghost = '[play_records[EXP_TYPE_GHOST]]'")
```

Comment:
@optimumtact said it didn't need to be sanitized as the values are all internalized.